tests/unit/swagger/templateHelpers.spec.ts

```diff
@@ -172,34 +172,60 @@ describe('ValidationService', () => {
     });
 
     describe('Enum validate', () => {
+        const name = 'name';
+        type Enumeration = string[] | number[];
+        interface FieldErrors { [key: string]: any; }
+        let validateEnum: (value: any, enumeration: Enumeration, fieldErrors?: FieldErrors) => any;
+
+        beforeEach(() => {
+          const validationService = new ValidationService({});
+          validateEnum = (
+            value: any,
+            enumeration: Enumeration,
+            fieldErrors?: FieldErrors,
+          ) => validationService.validateEnum(
+            name, value, fieldErrors || {}, enumeration);
+        });
+
         it('should enum number value', () => {
-            const value = 1;
-            const result = new ValidationService({}).validateEnum('name', value, {}, ['0', '1'] as any);
+            const value = '1';
+            const result = validateEnum(value, ['0', '1']);
             expect(result).to.equal(value);
         });
 
         it('should enum string value', () => {
             const value = 'HELLO';
-            const result = new ValidationService({}).validateEnum('name', value, {}, ['HELLO'] as any);
+            const result = validateEnum(value, ['HELLO']);
             expect(result).to.equal(value);
         });
 
         it('should enum no member', () => {
             const error: any = {};
-            const name = 'name';
             const value = 'HI';
-            const result = new ValidationService({}).validateEnum(name, value, error, [] as any);
+            const result = validateEnum(value, [], error);
             expect(result).to.equal(undefined);
             expect(error[name].message).to.equal(`no member`);
         });
 
         it('should enum out of member', () => {
             const error: any = {};
-            const name = 'name';
             const value = 'SAY';
-            const result = new ValidationService({}).validateEnum(name, value, error, ['HELLO', 'HI'] as any);
+            const result = validateEnum(value, ['HELLO', 'HI'], error);
             expect(result).to.equal(undefined);
-            expect(error[name].message).to.equal(`should be one of the following; ['HELLO', 'HI']`);
+            expect(error[name].message).to.equal(
+              `should be one of the following; ['HELLO', 'HI']`);
+        });
+
+        it('should not discriminate between numeric and string enum values', () => {
+          function run(
+            val: string | number,
+            enumeration: string[] | number[],
+          ) {
+            const result = validateEnum(val, enumeration);
+            expect(result).to.equal(val);
+          }
+          run('1', [0, 1]);
```

Comment:
I have a bit of trouble understanding this. Is the idea that in your app you could have a numeric enum defined, but incoming values in query parameters can be strings and so that should be valid?

Reply:
@drdgvhbh this PR ultimately looks great. I do agree with @endor here that the intent of your test isn’t very clear. I just mean that it would be wonderful if you would add a comment into the test explaining to future contributors why it’s okay that a stringified number satisfies a numeric enum check. Because I’m a bit uncertain if it is okay. So any clarification that you can add to the code would be helpful.

Reply:
I'm not sure if its okay either. It does preserve existing functionality though.

Currently an enum `[1, 2, 3]` becomes `['1', '2', '3']` in swagger. The change would make it the same as `[1, 2, 3]`, but you wouldn't screw over the people were inputting `'1'`, `'2'`, `'3'` in their http requests. 

I'd say turn it into a separate issue that gets put in the the next minor release.